### PR TITLE
BlurSameSize

### PR DESCRIPTION
**Pull request by jocke1995**
Always blurring a 720 render target. To keep the performance if increasing resolution, and to  keep the bloom-effect to look the same across different resolutions.

### CODE REVIEW

File: StortSpelProjekt/Engine/src/Renderer/Bloom.cpp [hunk 1]
```diff
@@ -35,11 +49,16 @@ Bloom::Bloom(
...
+
 	for (unsigned int i = 0; i < 2; i++)
 	{
 		m_PingPongResources[i] = new PingPongResource(m_Resources[i], device, dh_CBV_UAV_SRV, &srvDesc, &uavDesc);
 	}
+
+	D3D12_RENDER_TARGET_VIEW_DESC rtvDesc = {};
+	rtvDesc.Format = DXGI_FORMAT_R16G16B16A16_FLOAT;
+	rtvDesc.ViewDimension = D3D12_RTV_DIMENSION_TEXTURE2D;
+	m_PingPongResources[0]->createRTV(device, m_BlurWidth, m_BlurHeight, dh_RTV, &rtvDesc);
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/Bloom.cpp]: Quick comment on why only 1 render target

File: StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	ID3D12DescriptorHeap* d3d12DescriptorHeap = descriptorHeap_CBV_UAV_SRV->GetID3D12DescriptorHeap();
+	commandList->SetDescriptorHeaps(1, &d3d12DescriptorHeap);
+
+	commandList->SetGraphicsRootDescriptorTable(RS::dtSRV, descriptorHeap_CBV_UAV_SRV->GetGPUHeapAt(0));
+
+	// Change state on front/backbuffer
+	//commandList->ResourceBarrier(1, &CD3DX12_RESOURCE_BARRIER::Transition(
+	//	m_pDestinationRTV->GetResource()->GetID3D12Resource1(),
+	//	D3D12_RESOURCE_STATE_NON_PIXEL_SHADER_RESOURCE,
+	//	D3D12_RESOURCE_STATE_RENDER_TARGET));
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp]: unused code

File: StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	commandList->IASetIndexBuffer(m_pFullScreenQuadMesh->GetIndexBufferView());
+
+	// Draw a fullscreen quad 
+	commandList->DrawIndexedInstanced(m_NumIndices, 1, 0, 0, 0);
+
+	// Change state on front/backbuffer
+	//commandList->ResourceBarrier(1, &CD3DX12_RESOURCE_BARRIER::Transition(
+	//	m_pDestinationRTV->GetResource()->GetID3D12Resource1(),
+	//	D3D12_RESOURCE_STATE_RENDER_TARGET,
+	//	D3D12_RESOURCE_STATE_NON_PIXEL_SHADER_RESOURCE));
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp]: unused code

File: StortSpelProjekt/Engine/src/Renderer/DX12Tasks/RenderTask.cpp [hunk 4]
```diff
@@ -39,7 +39,7 @@ PipelineState* RenderTask::GetPipelineState(unsigned int index)
 	return m_PipelineStates[index];
 }
 
-void RenderTask::AddRenderTarget(std::string name, const RenderTargetView* renderTarget)
+void RenderTask::AddRenderTargetView(std::string name, const RenderTargetView* renderTarget)
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/DX12Tasks/RenderTask.cpp]: RenderTargetView

File: StortSpelProjekt/Engine/src/Renderer/DX12Tasks/RenderTask.cpp [hunk 5]
```diff
@@ -39,7 +39,7 @@ PipelineState* RenderTask::GetPipelineState(unsigned int index)
...
 {
 	m_RenderTargets[name] = renderTarget;
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/DX12Tasks/RenderTask.cpp]: RenderTargetView

File: StortSpelProjekt/Engine/src/Renderer/DX12Tasks/RenderTask.h [hunk 6]
```diff
@@ -41,7 +41,7 @@ class RenderTask : public DX12Task
 	PipelineState* GetPipelineState(unsigned int index);
 
 	
-	void AddRenderTarget(std::string, const RenderTargetView* renderTarget);
+	void AddRenderTargetView(std::string, const RenderTargetView* renderTarget);
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/DX12Tasks/RenderTask.h]: view

File: StortSpelProjekt/Engine/src/Renderer/HLSL/DownSamplePixel.hlsl [hunk 7]
```diff
@@ -0,0 +1,21 @@
+#include "../../Headers/structs.h"
+
+struct VS_OUT
+{
+	float4 pos  : SV_Position;
+	float2 uv   : UV;
+};
+
+// Source & Destination descriptorHeapIndices are stored in metallic & albedo
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/HLSL/DownSamplePixel.hlsl]: Metallic not used, correct comment

File: StortSpelProjekt/Engine/src/Renderer/PingPongResource.h [hunk 8]
```diff
@@ -27,11 +29,21 @@ class PingPongResource
 	const Resource* const GetResource() const;
 	const ShaderResourceView* const GetSRV() const;
 	const UnorderedAccessView* const GetUAV() const;
+	const RenderTargetView* const GetRTV() const;
 
 private:
+	friend class Bloom;
```
Comment (fjz345) [inline StortSpelProjekt/Engine/src/Renderer/PingPongResource.h]: No friend class

**Review by fjz345 — CHANGES_REQUESTED** [review]
Changes to be made

File: StortSpelProjekt/Engine/src/Renderer/Bloom.cpp (hunk 1 above)
Comment (jocke1995) [reply StortSpelProjekt/Engine/src/Renderer/Bloom.cpp]: We talked about this. The first pingPongResource acts as the "startingpoint" to read from. And the data comes from the output from brightTarget. Hence the need for a rtv

File: StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp (hunk 2 above)
Comment (jocke1995) [reply StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp]: Removed

File: StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp (hunk 3 above)
Comment (jocke1995) [reply StortSpelProjekt/Engine/src/Renderer/DX12Tasks/DownSampleRenderTask.cpp]: Removed

**Review by fjz345 — APPROVED** [review]
F

**Review by fredrix96 — APPROVED** [review]
The code standard seems fine and the program runs as it should

**Pull request merged**